src/ai/backend/agent/agent.py

```diff
@@ -835,7 +836,9 @@ async def _handle_destroy_event(self, ev: ContainerLifecycleEvent) -> None:
                         await self.rescan_resource_usage()
                         if not ev.suppress_events:
                             await self.produce_event(
-                                KernelTerminatedEvent(ev.kernel_id, "already-terminated"),
+                                KernelTerminatedEvent(
```

Comment:
Please update type definition of `KernelTerminationEventArgs.reason` variable accordingly. `KernelTerminationEventArgs` is a superclass of `KernelTerminatingEvent` and `KernelTerminatedEvent` event class.